rs/interfaces/src/execution_environment.rs

```diff
@@ -650,6 +660,41 @@ pub trait SystemApi {
     /// Canister id of the executing canister.
     fn canister_id(&self) -> ic_types::CanisterId;
 
+    /// Returns the number of environment variables.
+    fn ic0_env_var_count(&self) -> HypervisorResult<usize>;
+
+    /// Returns the size of the environment variable name at the given index.
+    fn ic0_env_var_name_size(&self, index: usize) -> HypervisorResult<usize>;
+
+    /// Copies the environment variable name at the given index into memory.
+    fn ic0_env_var_name_copy(
+        &self,
+        index: usize,
+        dst: usize,
+        offset: usize,
+        size: usize,
+        heap: &mut [u8],
+    ) -> HypervisorResult<()>;
+
+    /// Returns the size of the value for the environment variable with the given name.
+    fn ic0_env_var_value_size(
+        &self,
+        name_src: usize,
+        name_size: usize,
+        heap: &[u8],
+    ) -> HypervisorResult<usize>;
+
+    /// Copies the value of the environment variable with the given name into memory.
```

Comment:
Let's also point out the trap conditions (name is too long, name is not UTF-8, env var with given name does not exist).